sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceValidator.java

```diff
@@ -0,0 +1,284 @@
+            RecurrenceConstants.RECURRENCE_PATTERN_DAYS_OF_WEEK);
+        final List<String> daysOfWeek = settings.getRecurrence().getPattern().getDaysOfWeek();
+        if (daysOfWeek == null || daysOfWeek.size() == 0) {
+            reason = RecurrenceConstants.REQUIRED_PARAMETER;
+            return false;
+        }
+
+        for (String dayOfWeek : daysOfWeek) {
+            try {
+                DayOfWeek.valueOf(dayOfWeek.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                reason = RecurrenceConstants.UNRECOGNIZED_VALUE;
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private boolean validateFirstDayOfWeek() {
+        paramName = String.format("%s.%s.%s", TIME_WINDOW_FILTER_SETTING_RECURRENCE, RecurrenceConstants.RECURRENCE_PATTERN,
+            RecurrenceConstants.RECURRENCE_PATTERN_FIRST_DAY_OF_WEEK);
+
+        final String firstDayOfWeek = settings.getRecurrence().getPattern().getFirstDayOfWeek();
+        try {
+            DayOfWeek.valueOf(firstDayOfWeek.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            reason = RecurrenceConstants.UNRECOGNIZED_VALUE;
+            return false;
+        }
+        return true;
+    }
+
+    private boolean validateEndDate() {
+        paramName = String.format("%s.%s.%s", TIME_WINDOW_FILTER_SETTING_RECURRENCE, RecurrenceConstants.RECURRENCE_RANGE,
+            RecurrenceConstants.RECURRENCE_RANGE_EDN_DATE);
+        if (settings.getRecurrence().getRange().getEndDate() == null) {
+            reason = RecurrenceConstants.REQUIRED_PARAMETER;
+            return false;
+        }
+
+        if (settings.getRecurrence().getRange().getEndDate().isBefore(settings.getStart())) {
+            reason = RecurrenceConstants.OUT_OF_RANGE;
+            return false;
+        }
+        return true;
+    }
+
+    private boolean validateNumberOfOccurrences() {
+        paramName = String.format("%s.%s.%s", TIME_WINDOW_FILTER_SETTING_RECURRENCE, RecurrenceConstants.RECURRENCE_RANGE,
+            RecurrenceConstants.RECURRENCE_RANGE_NUMBER_OF_OCCURRENCES);
+        if (settings.getRecurrence().getRange().getNumberOfRecurrences() < 1) {
+            reason = RecurrenceConstants.OUT_OF_RANGE;
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Check whether the duration is shorter than the minimum gap between recurrence of days of week.
+     *
+     * @param duration       The duration of time window.
+     * @param interval       The number of weeks between each occurrence.
+     * @param daysOfWeek     The days of the week when the recurrence will occur.
+     * @param firstDayOfWeek The first day of the week.
+     * @return True if the duration is compliant with days of week, false otherwise.
+     */
+    private boolean isDurationCompliantWithDaysOfWeek(Duration duration, int interval, List<String> daysOfWeek, String firstDayOfWeek) {
```

Comment:
Just to be verify, this method is checking to make sure if I have a time window open the first one closes before the next starts?

Reply:
This logic seems like it might not work around daylight savings time.